service-loadbalancer/service_loadbalancer.go

```diff
@@ -388,8 +391,9 @@ func (lbc *loadBalancerController) getServices() (httpSvc []service, tcpSvc []se
 				continue
 			}
 			newSvc := service{
-				Name: getServiceNameForLBRule(&s, servicePort.Port),
-				Ep:   ep,
+				Name:        getServiceNameForLBRule(&s, servicePort.Port),
+				Ep:          ep,
+				BackendPort: servicePort.TargetPort.IntVal,
```

Comment:
Named ports are a thing, so lets just get this right here even. Please isolate the port switch case in getEndpoints to something like getTargetPort(servicePort) and call it from both places.